commit d5eb108ff4a94b03a014e90e6213bf7bf4faa374 (refs/changes/12/3712/2)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-03-23T19:30:17+00:00 (1 year, 7 months ago)
    
    OS-6820 bhyve provisioning times out after OS-6720
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Dan McDonald <danmcd@joyent.com>

diff --git a/usr/src/cmd/bhyve/bhyverun.c b/usr/src/cmd/bhyve/bhyverun.c
index 56240356ae..ca7983edd2 100644
--- a/usr/src/cmd/bhyve/bhyverun.c
+++ b/usr/src/cmd/bhyve/bhyverun.c
@@ -1119,17 +1119,16 @@ main(int argc, char *argv[])
 	 */
 	fbsdrun_addcpu(ctx, BSP, BSP, rip);
 
-	/*
-	 * Head off to the main event dispatch loop
-	 */
-	mevent_dispatch();
-#ifndef __FreeBSD
+#ifndef __FreeBSD__
 	if (vm_started_cb != NULL) {
 		vm_started_cb();
 	}
 #endif
 
-	pthread_exit(NULL);
+	/*
+	 * Head off to the main event dispatch loop
+	 */
+	mevent_dispatch();
 
 	exit(1);
 }
